src/InSAR/InSAR.jl

```diff
@@ -0,0 +1,94 @@
+"""
+function phase_ramp(rows::Vector{T}, columns::Vector{T}, burst_number::Int64, v_s::Float64,
+                    k_psi::Float64, dc_coefficient::Vector{Float64}, 
+                    dc_tau_0::Float64, fm_coefficient::Vector{Float64}, 
+                    fm_tau_0::Float64, f_c::Float64, lines_per_burst::Int64, 
+                    number_of_samples::Int64, delta_t_s::Float64, 
+                    delta_tau_s::Float64, tau_0::Number, c=SPEED_OF_LIGHT_CONSTANT::Real) where T <: Integer
+
+    tau = tau_0 .+ (columns .- 1) .* delta_tau_s # Slant range time of ith sample, Eqn. 12
+
+    # Doppler rate equations
+    k_s = 2 * v_s/c * f_c * k_psi; # Doppler rate from antenna scanning, Eqn. 4
+    alpha = 1 .- k_s ./ _doppler_fm_rate(tau, fm_coefficient, fm_tau_0); # conversion factor, Eqn. 3
+    k_t = k_s ./ alpha; # Doppler Centroid Rate, Eqn. 2
+
+    # Doppler azimuth time equations
+    eta_c = - _doppler_centroid_frequency(tau, dc_coefficient, dc_tau_0) ./ _doppler_fm_rate(tau, fm_coefficient, fm_tau_0); # Beam centre crossing time, Eqn. 7
+    tau_mid = tau_0 + number_of_samples/2 * delta_tau_s
+
+    eta_ref = eta_c .- (- _doppler_centroid_frequency(tau_mid, dc_coefficient, dc_tau_0) / _doppler_fm_rate(tau_mid, fm_coefficient, fm_tau_0)); # Reference time, Eqn. 6
+    line_in_burst = rows .- lines_per_burst * (burst_number - 1)
+    eta = -lines_per_burst / 2 * delta_t_s .+ (line_in_burst .- 1/2 ) .* delta_t_s
+
+    # the modulation term
+    doppler =  _doppler_centroid_frequency(tau, dc_coefficient, dc_tau_0)
+
+    # Compute the phase ramp added with the modulation term
+    ramp = [pi * k_t[j] * (eta[i] - eta_ref[j])^2 + 2 * pi * doppler[j] * (eta[i] - eta_ref[j]) for i=1:length(rows), j=1:length(columns)];
+    return ramp
+end
+
+"""
+phase_ramp(rows::AbstractArray{Int64, 2}, columns::AbstractArray{Int64, 2}, 
+            burst_number::Int64, meta_data::Sentinel1MetaData, 
+            orbit_states::Vector{SARProcessing.OrbitState})
+
+Extracts relevant parameters from meta_data and calls phase_ramp().
+"""
+function phase_ramp(rows::Vector{T}, columns::Vector{T}, 
+                        burst_number::Int64, v_s::Float64, meta_data::Sentinel1MetaData) where T <: Integer
+
+    k_psi = meta_data.product.azimuth_steering_rate * pi/180 ;
+    dc_coefficient = meta_data.bursts[burst_number].doppler_centroid.polynomial;
+    dc_tau_0 = meta_data.bursts[burst_number].doppler_centroid.t0;
+    fm_coefficient = meta_data.bursts[burst_number].azimuth_fm_rate.polynomial;
+    fm_tau_0 = meta_data.bursts[burst_number].azimuth_fm_rate.t0;
+    f_c = meta_data.product.radar_frequency;
+    lines_per_burst = meta_data.swath.lines_per_burst;
+    number_of_samples = meta_data.image.number_of_samples;
+    delta_t_s = meta_data.image.azimuth_time_interval;
+    delta_tau_s = 1/meta_data.product.range_sampling_rate
+    tau_0 = meta_data.image.slant_range_time_seconds; # in seconds
+    c = 299792458;
```

Comment:
We should change this so we have the `c` as a constant. Probably be easier to do in another pr after the merge